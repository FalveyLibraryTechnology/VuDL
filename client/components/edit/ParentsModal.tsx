import React, { useEffect, useState } from "react";
import Checkbox from "@mui/material/Checkbox";
import Dialog from "@mui/material/Dialog";
import DialogContent from "@mui/material/DialogContent";
import DialogTitle from "@mui/material/DialogTitle";
import FormControl from "@mui/material/FormControl";
import FormControlLabel from "@mui/material/FormControlLabel";
import FormLabel from "@mui/material/FormLabel";
import Grid from "@mui/material/Grid";
import Radio from "@mui/material/Radio";
import RadioGroup from "@mui/material/RadioGroup";
import IconButton from "@mui/material/IconButton";
import CloseIcon from "@mui/icons-material/Close";
import { useEditorContext } from "../../context/EditorContext";
import { getObjectRecursiveChildPidsUrl, getObjectStateUrl } from "../../util/routes";
import { useFetchContext } from "../../context/FetchContext";
import ObjectLoader from "./ObjectLoader";

const ParentsModal = (): React.ReactElement => {
    const {
        state: { isParentsModalOpen, objectDetailsStorage, parentsModalActivePid },
        action: { removeFromObjectDetailsStorage, setSnackbarState, toggleParentsModal },
    } = useEditorContext();
    const {
        action: { fetchJSON, fetchText },
    } = useFetchContext();
    const [statusMessage, setStatusMessage] = useState<string>("");
    const [includeChildren, setIncludeChildren] = useState<boolean>(false);
    const [selectedValue, setSelectedValue] = useState<string>("Inactive");
    const [childPidResponse, setChildPidResponse] = useState({ loading: true });
    const loaded = Object.prototype.hasOwnProperty.call(objectDetailsStorage, parentsModalActivePid);
    const details = loaded ? objectDetailsStorage[parentsModalActivePid] : {};

    const showSnackbarMessage = (message: string, severity: string) => {
        setSnackbarState({
            open: true,
            message,
            severity,
        });
    };

    const contents = "Coming soon...";

    return (
        <Dialog className="parentsModal" open={isParentsModalOpen} onClose={toggleParentsModal} fullWidth={true}>
            <DialogTitle>
                <Grid container>
                    <Grid item xs={11}>
                        Parents Editor ({parentsModalActivePid})
                    </Grid>
                    <Grid item xs={1}>
                        <IconButton className="closeButton" onClick={toggleParentsModal}>
                            <CloseIcon />
                        </IconButton>
                    </Grid>
                </Grid>
            </DialogTitle>
            <DialogContent>
                {parentsModalActivePid ? <ObjectLoader pid={parentsModalActivePid} /> : ""}
                {contents}
            </DialogContent>
        </Dialog>
    );
};

export default ParentsModal;
