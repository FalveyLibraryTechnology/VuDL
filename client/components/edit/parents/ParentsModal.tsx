import React from "react";
import Dialog from "@mui/material/Dialog";
import DialogContent from "@mui/material/DialogContent";
import DialogTitle from "@mui/material/DialogTitle";
import Grid from "@mui/material/Grid";
import IconButton from "@mui/material/IconButton";
import CloseIcon from "@mui/icons-material/Close";
import { useGlobalContext } from "../../../context/GlobalContext";
import { useEditorContext } from "../../../context/EditorContext";
import ObjectLoader from "../ObjectLoader";
import ParentList from "./ParentList";
import ParentPicker from "./ParentPicker";

const ParentsModal = (): React.ReactElement => {
    const {
        state: { parentsModalActivePid },
        action: { isModalOpen, openModal, closeModal },
    } = useGlobalContext();
    const {
        state: { objectDetailsStorage },
    } = useEditorContext();

    const loaded = Object.prototype.hasOwnProperty.call(objectDetailsStorage, parentsModalActivePid);

    const contents = (
        <>
            <h3>Add New Parent:</h3>
            <ParentPicker pid={parentsModalActivePid} />
            <h3>Parents</h3>
            {parentsModalActivePid ? <ParentList pid={parentsModalActivePid} /> : ""}
        </>
    );
    return (
        <Dialog className="parentsModal" open={isModalOpen("parents")} onClose={closeModal} fullWidth={true}>
            <DialogTitle>
                <Grid container>
                    <Grid item xs={11}>
                        Parents Editor ({parentsModalActivePid})
                    </Grid>
                    <Grid item xs={1}>
                        <IconButton className="closeButton" onClick={closeModal}>
                            <CloseIcon />
                        </IconButton>
                    </Grid>
                </Grid>
            </DialogTitle>
            <DialogContent>
                {parentsModalActivePid ? <ObjectLoader pid={parentsModalActivePid} /> : ""}
                {loaded ? contents : ""}
            </DialogContent>
        </Dialog>
    );
};

export default ParentsModal;
