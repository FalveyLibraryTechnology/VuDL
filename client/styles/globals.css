:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
