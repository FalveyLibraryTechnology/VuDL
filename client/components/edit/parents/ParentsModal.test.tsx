import React from "react";
import { describe, beforeEach, expect, it, jest } from "@jest/globals";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import ParentsModal from "./ParentsModal";

const mockUseEditorContext = jest.fn();
jest.mock("../../../context/EditorContext", () => ({
    useEditorContext: () => {
        return mockUseEditorContext();
    },
}));
const mockUseGlobalContext = jest.fn();
jest.mock("../../../context/GlobalContext", () => ({
    useGlobalContext: () => {
        return mockUseGlobalContext();
    },
}));
jest.mock("../ObjectLoader", () => () => "ObjectLoader");
jest.mock("./ParentList", () => () => "ParentList");
jest.mock("./ParentPicker", () => () => "ParentPicker");

describe("ParentsModal", () => {
    let editorValues;
    let globalValues;
    let pid: string;
    beforeEach(() => {
        pid = "foo:123";
        editorValues = {
            state: {
                objectDetailsStorage: {},
                isParentsModalOpen: true,
                parentsModalActivePid: pid,
            },
        };
        mockUseEditorContext.mockReturnValue(editorValues);
        globalValues = {
            action: {
                closeModal: jest.fn(),
                isModalOpen: jest.fn(),
            }
        }
        mockUseGlobalContext.mockReturnValue(globalValues);
        globalValues.action.isModalOpen.mockReturnValue(true);
    });

    it("renders correctly for a non-loaded PID", () => {
        const wrapper = shallow(<ParentsModal />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(globalValues.action.isModalOpen).toHaveBeenCalledWith("parents");
    });

    it("renders correctly for a loaded PID", () => {
        editorValues.state.objectDetailsStorage[pid] = { pid };
        const wrapper = shallow(<ParentsModal />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(globalValues.action.isModalOpen).toHaveBeenCalledWith("parents");
    });

    it("renders correctly when PID is unset", () => {
        editorValues.state.parentsModalActivePid = null;
        const wrapper = shallow(<ParentsModal />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(globalValues.action.isModalOpen).toHaveBeenCalledWith("parents");
    });

    it("toggles the modal", () => {
        const component = mount(<ParentsModal />);
        component.find("button").simulate("click");

        expect(globalValues.action.closeModal).toHaveBeenCalledWith("parents");
        component.unmount();
    });
});
