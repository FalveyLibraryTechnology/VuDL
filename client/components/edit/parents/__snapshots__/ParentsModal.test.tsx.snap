// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentsModal renders correctly for a loaded PID 1`] = `
<ForwardRef(Dialog)
  className="parentsModal"
  fullWidth={true}
  onClose={[Function]}
  open={true}
>
  <ForwardRef(DialogTitle)>
    <ForwardRef(Grid)
      container={true}
    >
      <ForwardRef(Grid)
        item={true}
        xs={11}
      >
        Parents Editor (
        foo:123
        )
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        xs={1}
      >
        <ForwardRef(IconButton)
          className="closeButton"
          onClick={[Function]}
        >
          <Memo(ForwardRef(CloseIcon)) />
        </ForwardRef(IconButton)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <Component
      pid="foo:123"
    />
    <h3>
      Add New Parent:
    </h3>
    <Component
      pid="foo:123"
    />
    <h3>
      Parents
    </h3>
    <Component
      pid="foo:123"
    />
  </ForwardRef(DialogContent)>
</ForwardRef(Dialog)>
`;

exports[`ParentsModal renders correctly for a non-loaded PID 1`] = `
<ForwardRef(Dialog)
  className="parentsModal"
  fullWidth={true}
  onClose={[Function]}
  open={true}
>
  <ForwardRef(DialogTitle)>
    <ForwardRef(Grid)
      container={true}
    >
      <ForwardRef(Grid)
        item={true}
        xs={11}
      >
        Parents Editor (
        foo:123
        )
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        xs={1}
      >
        <ForwardRef(IconButton)
          className="closeButton"
          onClick={[Function]}
        >
          <Memo(ForwardRef(CloseIcon)) />
        </ForwardRef(IconButton)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent)>
    <Component
      pid="foo:123"
    />
  </ForwardRef(DialogContent)>
</ForwardRef(Dialog)>
`;

exports[`ParentsModal renders correctly when PID is unset 1`] = `
<ForwardRef(Dialog)
  className="parentsModal"
  fullWidth={true}
  onClose={[Function]}
  open={true}
>
  <ForwardRef(DialogTitle)>
    <ForwardRef(Grid)
      container={true}
    >
      <ForwardRef(Grid)
        item={true}
        xs={11}
      >
        Parents Editor (
        )
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
        xs={1}
      >
        <ForwardRef(IconButton)
          className="closeButton"
          onClick={[Function]}
        >
          <Memo(ForwardRef(CloseIcon)) />
        </ForwardRef(IconButton)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(DialogTitle)>
  <ForwardRef(DialogContent) />
</ForwardRef(Dialog)>
`;
